Stringifying a mocked object returns undefined

Hey! Using `JSON.stringify` on a mocked object returns `undefined`, is that expected?

```ts
interface Test {
    test: string
    hello: number
}

const mocked = mock<Test>({ test: 'default' })
console.log(mocked) // { test: 'default', _isMockObject: true }
console.log(JSON.stringify(mocked)) // undefined
```

Shouldn't this return `'{"test":"default"}'` or at least `'{"test":"default","_isMockObject":true}'`?

Some details:
```ts
interface Test {
  cookie: {
    domain: string
  }
}
```
---
This code:
```ts
test.only('dummy test 1', () => {
  const cookie = { domain: 'dummy' }
  const data = mock<Test>({ cookie })

  console.log('cookie before stringifying data', cookie)
  JSON.stringify(data)
  console.log('cookie after stringifying data', cookie)

  expect(1).toBe(1)
})
```
outputs this:
```
cookie before stringifying data { domain: 'dummy' }
cookie after stringifying data {
  domain: 'dummy',
  toJSON: [Function: spy] {
    getMockName: [Function (anonymous)],
    mockName: [Function (anonymous)],
    mockClear: [Function (anonymous)],
    mockReset: [Function (anonymous)],
    mockRestore: [Function (anonymous)],
    getMockImplementation: [Function (anonymous)],
    mockImplementation: [Function (anonymous)],
    mockImplementationOnce: [Function (anonymous)],
    withImplementation: [Function: withImplementation],
    mockReturnThis: [Function (anonymous)],
    mockReturnValue: [Function (anonymous)],
    mockReturnValueOnce: [Function (anonymous)],
    mockResolvedValue: [Function (anonymous)],
    mockResolvedValueOnce: [Function (anonymous)],
    mockRejectedValue: [Function (anonymous)],
    mockRejectedValueOnce: [Function (anonymous)],
    calledWith: [Function (anonymous)]
  }
}
```

Note that I stringify `data` and not `cookie`.

---
And this one:
```ts
test.only('dummy test 2', () => {
  const cookie = { domain: 'dummy' }
  const data = mock<Test>({ cookie })

  JSON.stringify(data)

  expect({
    cookie: {
      domain: 'dummy',
    },
  }).toStrictEqual({ cookie })
})
```
outputs this:
```shell
AssertionError: expected { cookie: { domain: 'dummy' } } to strictly equal { cookie: { domain: 'dummy', …(1) } }

- Expected
+ Received

  Object {
-   "cookie": undefined,
+   "cookie": Object {
+     "domain": "dummy",
+   },
  }

 ❯ src/...:86:6
     84|       domain: 'dummy',
     85|     },
     86|   }).toStrictEqual({ cookie })
       |      ^
     87| })
     88|
```
While if I remove the stringify it passes:
```ts
test.only('dummy test 3', () => {
  const cookie = { domain: 'dummy' }
  const data = mock<Test>({ cookie })

  // JSON.stringify(data)

  expect({
    cookie: {
      domain: 'dummy',
    },
  }).toStrictEqual({ cookie })
})
```

It seems there is some edge effects.

--- a/src/Mock.spec.ts
+++ b/src/Mock.spec.ts
@@ -651,42 +651,21 @@ describe('vitest-mock-extended', () => {
             expect(stringified).toStrictEqual('{"test":"default","_isMockObject":true}');
         });
 
-        //     test('should strictly equal', () => {
-        //         interface Test {
-        //             id: string;
-        //             cookie: string;
-        //         }
-        //         const data = mock<Test>({
-        //             id: 'test',
-        //             cookie: 'test',
-        //         });
-        //
-        //         const receiveSomehow = (data: Test) => ({
-        //             data: {
-        //                 id: data.id,
-        //                 cookie: data.cookie,
-        //             },
-        //         });
-        //
-        //         const receivedSomehow = receiveSomehow(data);
-        //
-        //         expect(receivedSomehow).toStrictEqual({ data });
-        //     });
-        //
-        //     test('should not modify the mock', () => {
-        //         interface Test {
-        //             cookie: {
-        //                 domain: string;
-        //             };
-        //         }
-        //         const cookie = { domain: 'dummy' };
-        //         const data = mock<Test>({ cookie });
-        //
-        //         console.log('cookie before stringifying data', cookie);
-        //         JSON.stringify(data);
-        //         console.log('cookie after stringifying data', cookie);
-        //
-        //         expect(1).toBe(1);
-        //     });
+        test('should stricktly equal after stringify', () => {
+            //given
+            interface Test {
+                cookie: {
+                    domain: string;
+                };
+            }
+            const cookie = { domain: 'dummy' };
+            const data = mock<Test>({ cookie });
+
+            //when
+            JSON.stringify(data);
+
+            //when
+            expect({ cookie: { domain: 'dummy' } }).toStrictEqual({ cookie });
+        });
     });
 });

--- a/src/Mock.ts
+++ b/src/Mock.ts
@@ -128,13 +128,11 @@ const handler = (opts?: MockOpts) => ({
     },
 
     set: (obj: MockProxy<any>, property: ProxiedProperty, value: any) => {
-        // @ts-ignore All of these ignores are due to https://github.com/microsoft/TypeScript/issues/1863
         obj[property] = value;
         return true;
     },
 
     get: (obj: MockProxy<any>, property: ProxiedProperty) => {
-        // @ts-ignore
         if (!(property in obj)) {
             if (property === '_isMockObject' || property === '_isMockFunction') {
                 return undefined;
@@ -145,21 +143,22 @@ const handler = (opts?: MockOpts) => ({
             }
             // Jest's internal equality checking does some wierd stuff to check for iterable equality
             if (property === Symbol.iterator) {
-                // @ts-ignore
                 return obj[property];
             }
+
+            if (property === 'toJSON') {
+                return JSON.stringify(obj);
+            }
+
             // So this calls check here is totally not ideal - jest internally does a
             // check to see if this is a spy - which we want to say no to, but blindly returning
             // an proxy for calls results in the spy check returning true. This is another reason
             // why deep is opt in.
             const fn = calledWithFn({ fallbackMockImplementation: opts?.fallbackMockImplementation });
             if (opts?.deep && property !== 'calls') {
-                // @ts-ignore
                 obj[property] = new Proxy<MockProxy<any>>(fn, handler(opts));
-                // @ts-ignore
                 obj[property]._isMockObject = true;
             } else {
-                // @ts-ignore
                 obj[property] = fn;
             }
         }
@@ -170,11 +169,6 @@ const handler = (opts?: MockOpts) => ({
             return obj[property].bind(obj);
         }
 
-        if (property === 'toJSON') {
-            return JSON.stringify(obj);
-        }
-
-        // @ts-ignore
         return obj[property];
     },
 });

